Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -611,14 +614,100 @@ function vtkOpenGLTexture(publicAPI, model) {
   };
 
   //----------------------------------------------------------------------------
-  function updateArrayDataType(dataType, data, depth = false) {
+
+  /**
+   * Reads a region from the image data and writes to the given output array.
+   *
+   * Assumes X varies the fastest and Z varies the slowest.
+   *
+   * @param {*} data
+   * @param {*} start
+   * @param {*} size
+   * @param {*} outArray
+   * @param {*} outOffset
+   * @returns
+   */
+  function readRegion(data, start, size, outArray, outOffset) {
+    const [sx, sy, sz] = size;
+    const sxy = sx * sy;
+    const startOffset = start[2] * sxy + start[1] * sx + start[0];
+
+    let writeOffset = outOffset;
+    for (let zi = 0; zi < sz; zi++) {
+      const zOffset = startOffset + zi * sxy;
+      for (let yi = 0; yi < sy; yi++) {
+        const readOffset = zOffset + yi * sx;
+        const chunk = data.subarray(readOffset, readOffset + sx);
```

Comment:
did you check the performances ? 
`subarray()` is somehow costly, I found out that replacing it with a `for` loop  is sometimes faster.

Reply:
I have not, but I do recall subarray in v8 being much slower than usual. Best be safe if that's still an issue.